pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -98,6 +104,14 @@ func secretMissingMetric(namespace, name string) map[string]string {
 	}
 }
 
+func getHostFromAPIURL(apiURL string) (string, error) {
```

Comment:
We will require this function to extract the hostname (which will always have the domain at the end) because if its just API URL, it will end with `:6443` so `dns.IsManaged()` will return a `false` every time as it checks for string ending with managed domain name.